src/com/seafile/seadroid2/ui/activity/SearchActivity.java

```diff
@@ -0,0 +1,430 @@
+            @Override
+            public void onLoadMore(int page, int totalItemsCount) {
+                // Triggered only when new data needs to be appended to the list
+                // Add whatever code is needed to append new items to your AdapterView
+                // handleSearch(totalItemsCount);
+            }
+        });
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        initData();
+    }
+
+    @Override
+    protected void onDestroy() {
+        Log.d(DEBUG_TAG, "onDestroy is called");
+        if (txService != null) {
+            unbindService(mConnection);
+            txService = null;
+        }
+
+        super.onDestroy();
+    }
+
+    private void initData() {
+        AccountManager accountManager = new AccountManager(this);
+        account = accountManager.getCurrentAccount();
+        dataManager = new DataManager(account);
+
+        // bind transfer service
+        Intent bIntent = new Intent(this, TransferService.class);
+        bindService(bIntent, mConnection, Context.BIND_AUTO_CREATE);
+        Log.d(DEBUG_TAG, "try bind TransferService");
+    }
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if (mSearchedFiles.isEmpty()) {
+            mMessageContainer.setVisibility(View.VISIBLE);
+        } else {
+            mMessageContainer.setVisibility(View.GONE);
+        }
+
+    }
+
+    @Override
+    public void onClick(View v) {
+        final int id = v.getId();
+        if (id == R.id.btn_search) {
+            // pass 0 to disable page loading
+            handleSearch(0);
+        } else if (id == R.id.btn_clear) {
+            mTextField.getText().clear();
+        }
+
+    }
+
+    private void handleSearch(int page) {
+        // server only supports 100 search result right now
+        page = 100;
+        if (!Utils.isNetworkOn()) {
+            showToast(R.string.network_down);
+            mMessageContainer.setVisibility(View.VISIBLE);
+            return;
+        } else
+            mMessageContainer.setVisibility(View.GONE);
+
+        String searchText = mTextField.getText().toString().trim();
+        if (!TextUtils.isEmpty(searchText)) {
+            // mSearchedFiles.clear();
+
+            search(searchText, page);
+
+            Utils.hideSoftKeyboard(mTextField);
+        } else {
+            showToast(R.string.search_txt_empty);
+        }
+    }
+
+    private void search(String content, int page) {
+        // get current account
+        AccountManager accountManager = new AccountManager(this);
+        Account account = accountManager.getCurrentAccount();
+        DataManager dataManager = new DataManager(account);
```

Comment:
We already has a `DataManager` member in the class, so why not use it but create a new  `DataManager` object each time?